wpilibj/src/test/java/edu/wpi/first/wpilibj/shuffleboard/ShuffleboardTest.java

```diff
@@ -0,0 +1,337 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2016-2018 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.shuffleboard;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.networktables.NetworkTableType;
+import edu.wpi.first.wpilibj.Sendable;
+import edu.wpi.first.wpilibj.SendableBase;
+import edu.wpi.first.wpilibj.UnitTestUtility;
+import edu.wpi.first.wpilibj.UtilityClassTest;
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static edu.wpi.first.networktables.NetworkTableType.kBoolean;
+import static edu.wpi.first.networktables.NetworkTableType.kDouble;
+import static edu.wpi.first.networktables.NetworkTableType.kString;
+import static edu.wpi.first.wpilibj.shuffleboard.Shuffleboard.add;
+
+@SuppressWarnings("PMD.TooManyMethods")
+public class ShuffleboardTest extends UtilityClassTest {
+
+  private static NetworkTableInstance instance;
+
+  public ShuffleboardTest() {
+    super(Shuffleboard.class);
+  }
+
+  @BeforeAll
+  public static void setUpClass() {
+    UnitTestUtility.setupMockBase();
+  }
+
+  @BeforeEach
+  public void setUp() {
+    instance = NetworkTableInstance.create();
+    Shuffleboard.setNtInstance(instance);
+  }
+
+  @AfterAll
+  public static void tearDown() {
+    Shuffleboard.setNtInstance(NetworkTableInstance.getDefault());
+  }
+
+  @Test
+  public void testInvalidInputsToAddEntry() {
+    assertAll(
+        () -> assertThrows(NullPointerException.class, () -> add(null, kDouble)),
+        () -> assertThrows(NullPointerException.class, () -> add("foo", (NetworkTableType) null)),
+        () -> assertThrows(IllegalArgumentException.class, () -> add("", kDouble)),
+        () -> assertThrows(IllegalArgumentException.class, () -> add("a/b", kDouble)));
+  }
```

Comment:
One test per test case please.  If you want them grouped use nested test cases.

Reply:
Ooh, that's new. Would you recommend grouping entries vs sendables, or have something more finely grained?

Reply:
I would make this a parameterized test called `invalidInputsToAddEntryTest`.  Parameters would be the expected exception, the key for the add method, and the object for the add method.